From d90eb5f9c4afd3881b7cb066ea3a06e693803095 Mon Sep 17 00:00:00 2001
From: Thao Luong <thao.luong.uw@rvc.renesas.com>
Date: Wed, 13 May 2015 18:30:42 +0700
Subject: [PATCH 1/1] omxaudiodec: Fix issue memory leak of omxaacdec due to
 unmap input buffer

This patch file will unmap input buffer after usage

https://bugzilla.gnome.org/show_bug.cgi?id=736314

refs #69205
Signed-off-by: Thao Luong <thao.luong.uw@rvc.renesas.com>
---
 omx/gstomxaudiodec.c                    |   8 ++++++++
 1 files changed, 8 insertions(+)

diff --git a/omx/gstomxaudiodec.c b/omx/gstomxaudiodec.c
index ff8ea53..a6577b2 100644
--- a/omx/gstomxaudiodec.c
+++ b/omx/gstomxaudiodec.c
@@ -1097,6 +1097,7 @@ gst_omx_audio_dec_handle_frame (GstAudioDecoder * decoder, GstBuffer * inbuf)
       gst_omx_port_release_buffer (port, buf);
       goto flow_error;
     }
+    gst_buffer_unmap (inbuf, &minfo);
 
     if (self->codec_data) {
       GST_DEBUG_OBJECT (self, "Passing codec data to the component");
@@ -1187,6 +1188,7 @@ gst_omx_audio_dec_handle_frame (GstAudioDecoder * decoder, GstBuffer * inbuf)
 
 full_buffer:
   {
+    gst_buffer_unmap (inbuf, &minfo);
     if (inbuf)
       gst_buffer_unref (inbuf);
 
@@ -1198,6 +1200,7 @@ full_buffer:
 
 flow_error:
   {
+    gst_buffer_unmap (inbuf, &minfo);
     if (inbuf)
       gst_buffer_unref (inbuf);
 
@@ -1206,6 +1209,7 @@ flow_error:
 
 too_large_codec_data:
   {
+    gst_buffer_unmap (inbuf, &minfo);
     if (inbuf)
       gst_buffer_unref (inbuf);
 
@@ -1218,6 +1222,7 @@ too_large_codec_data:
 
 component_error:
   {
+    gst_buffer_unmap (inbuf, &minfo);
     if (inbuf)
       gst_buffer_unref (inbuf);
 
@@ -1230,6 +1235,7 @@ component_error:
 
 flushing:
   {
+    gst_buffer_unmap (inbuf, &minfo);
     if (inbuf)
       gst_buffer_unref (inbuf);
 
@@ -1238,6 +1244,7 @@ flushing:
   }
 reconfigure_error:
   {
+    gst_buffer_unmap (inbuf, &minfo);
     if (inbuf)
       gst_buffer_unref (inbuf);
 
@@ -1247,6 +1254,7 @@ reconfigure_error:
   }
 release_error:
   {
+    gst_buffer_unmap (inbuf, &minfo);
     if (inbuf)
       gst_buffer_unref (inbuf);
 
-- 
2.4.0

